### ISSUE TITLE
Functions wrapped in `__importDefault` with allowSyntheticDefaultImports emit incorrect function calls

### ISSUE THREAD

**Issue by borekb** [microsoft/TypeScript#35420]
The following code:

```ts
import allSettled from "promise.allsettled";

const resolved = Promise.resolve(42);
const rejected = Promise.reject(-1);

allSettled([resolved, rejected]).then(results => {
  console.log(results);
});
```

Is transpiled into the following JS with `esModuleInterop` & `allowSyntheticDefaultImports`:

```js
"use strict";
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, "__esModule", { value: true });
const promise_allsettled_1 = __importDefault(require("promise.allsettled"));
const resolved = Promise.resolve(42);
const rejected = Promise.reject(-1);
promise_allsettled_1.default([resolved, rejected]).then(results => {
  console.log(results);
});
```

Full `tsconfig.json`:

```js
{
  "compilerOptions": {
    "target": "es2018",
    "module": "commonjs",
    "strict": true,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
  },
  "include": [
    "src"
  ]
}
```

([CodeSandbox](https://codesandbox.io/s/inspiring-lewin-eyt6r).)

Executing this code leads to:

```
TypeError: #<Object> is not a constructor
    at Object.resolve (<anonymous>)
    at Object.PromiseResolve (/sandbox/node_modules/es-abstract/es2018.js:160:10)
    at /sandbox/node_modules/promise.allsettled/implementation.js:26:24
    at Function.from (<anonymous>)
    at Object.allSettled (/sandbox/node_modules/promise.allsettled/implementation.js:19:22)
    at Object.allSettled [as default] (/sandbox/node_modules/promise.allsettled/index.js:16:9)
    at Object.<anonymous> (/sandbox/src/demo.ts:6:11)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Module.m._compile (/sandbox/node_modules/ts-node/src/index.ts:536:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:787:10)
```

That is because the promise.allsettled package handles `this` according to the spec, see https://github.com/es-shims/Promise.allSettled/issues/5. The author of that library is @ljharb who will have deeper understanding of the specifics than I do. (Also, thanks for nudging me to report this issue!)

The problem is most likely the emitted function call, which looks like this:

```js
promise_allsettled_1.default([resolved, rejected]).then(results => {
  console.log(results);
});
```

In comparison, Babel + TS emits this (which works):

```js
(0, _promise.default)([resolved, rejected]).then(function (results) {
  console.log(results);
});
```

[Babel playground](https://babeljs.io/en/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=JYWwDg9gTgLgBAQwDZIMoFMYyegJnAMyghDgHIxiRgBndAOmSTqx1zIG4AoLgYwgB2NeFHQ0ISAG544AXjgAFKrQajxU9AAoALACYAlN35CR6AFbpeMGfKUkV9URauaAtAEZDPJhlZ5NANpqEtK4ADRwTpbWuAC6-vQwABboAppqAK5IMDRyAHxwAN5ccHDG6gxIEADm6WJZOV4AvoZAA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2%2Ctypescript&prettier=false&targets=&version=7.7.4&externalPlugins=).

**TypeScript Version:**  3.7.2

**Search Terms:** `esModuleInterop`, `allowSyntheticDefaultImports`, emit, transpile, downlevel compile, ES Modules, default, wrapper, Node.js, module.

**Comment by ljharb** [comment]
A simpler repro case would be importing `module.exports = function () { 'use strict'; return !!this; }` (with esModuleInterop and synthetic imports enabled, of course) - a correct downleveling would make invoking this function with no implied receiver return false; TS’s output currently seems to make it return true.

**Comment by ajafff** [comment]
AFAICT the same bug exists for named imports.

```ts
// foo.ts
export function foo() { return !!this; }

// bar.ts
import { foo } from './foo.js';
foo();

// transpiled bar.js
var foo_js_1 = require('./foo.js');
foo_js_1.foo(); // this line should be `(0, foo_js_1.foo)()`
```

@ljharb please correct me if I'm wrong.

**Comment by ljharb** [comment]
Yes, that’s the same bug, thanks.

**Comment by ajafff** [comment]
Fix is up at #35877

**Comment by gdh1995** [comment]
@rbuckton Sorry to disturb you, but your PR seems to wrap all imported symbols with `(0, ...)`, but I think it's not necessary and may be skipped when the target is a "const arrow function".
